homeassistant/components/fan/__init__.py

```diff
@@ -155,11 +179,22 @@ def is_on(self):
     @property
     def speed(self) -> Optional[str]:
         """Return the current speed."""
+        if self.supported_features & SUPPORT_SET_SPEED_PERCENTAGE:
+            return self.percentage_to_speed(self.percentage_speed)
```

Comment:
We shouldn't do conversion for both. 

I think that if we go for this, we should say that percentage is the base way of setting speed. If you only support different modes, we can provide helper methods to expose it as percentages (3 modes = 0%, 50% and 100%)

Reply:
The current helper matches ranges rather than specific points. So 0 = off, 1-33=low, 34-66=medium, 67+=high. Does that seem reasonable?

Reply:
Yeah that's fine too .